Sources/Basics/AsyncProcess.swift

```diff
@@ -0,0 +1,1466 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import _Concurrency
+import Dispatch
+import Foundation
+
+#if os(Windows)
+import TSCLibc
+#endif
+
+#if os(Linux)
+#if USE_IMPL_ONLY_IMPORTS
+@_implementationOnly
+import func TSCclibc.SPM_posix_spawn_file_actions_addchdir_np_supported
+
+@_implementationOnly
+import func TSCclibc.SPM_posix_spawn_file_actions_addchdir_np
+#else
+private import func TSCclibc.SPM_posix_spawn_file_actions_addchdir_np_supported
+private import func TSCclibc.SPM_posix_spawn_file_actions_addchdir_np
+#endif // #if USE_IMPL_ONLY_IMPORTS
+#endif // #if os(Linux)
+
+import class TSCBasic.CStringArray
+import class TSCBasic.LocalFileOutputByteStream
+import enum TSCBasic.ProcessEnv
+import struct TSCBasic.ProcessEnvironmentBlock
+import enum TSCBasic.SystemError
+import class TSCBasic.Thread
+import protocol TSCBasic.WritableByteStream
+
+/// Process result data which is available after process termination.
+package struct AsyncProcessResult: CustomStringConvertible, Sendable {
+    package enum Error: Swift.Error, Sendable {
+        /// The output is not a valid UTF8 sequence.
+        case illegalUTF8Sequence
+
+        /// The process had a non zero exit.
+        case nonZeroExit(AsyncProcessResult)
+
+        /// The process failed with a `SystemError` (this is used to still provide context on the process that was launched).
+        case systemError(arguments: [String], underlyingError: Swift.Error)
+    }
+
+    package enum ExitStatus: Equatable, Sendable {
+        /// The process was terminated normally with a exit code.
+        case terminated(code: Int32)
+#if os(Windows)
+        /// The process was terminated abnormally.
+        case abnormal(exception: UInt32)
+#else
+        /// The process was terminated due to a signal.
+        case signalled(signal: Int32)
+#endif
+    }
+
+    /// The arguments with which the process was launched.
+    package let arguments: [String]
+
+    /// The environment with which the process was launched.
+    package let environmentBlock: ProcessEnvironmentBlock
+
+    @available(*, deprecated, renamed: "env")
+    package var environment: [String:String] {
```

Comment:
Nit: I think all other spots I saw for dictionaries had a space between key and value type, is this a formatter problem?

Reply:
I forgot to format that file, cleaning up now...